### PR TITLE
Add additional resource descriptions

### PR DESCRIPTION
**Pull request by tas50**
These are straight from our docs when available.

### CODE REVIEW

File: lib/chef/resource/execute.rb [hunk 1]
```diff
@@ -55,28 +55,53 @@ def command(arg = nil)
...
+               description: "When true this enables ENV magic to add path_sanity to the PATH and force the locale to English+UTF-8 for parsing output"
+
+      property :returns, [ Integer, Array ], default: 0,
+               description: "The return value for a command. This may be an array of accepted values. An exception is raised when the return value(s) do not match."
+
+      property :timeout, [ Integer, Float ],
+               description: "The amount of time (in seconds) a command is to wait before timing out."
+
+      property :user, [ String, Integer ],
+               description: "The user name of the user identity with which to launch the new process. Default value: nil. The user name may optionally be specifed with a domain, i.e. domainuser or user@my.dns.domain.com via Universal Principal Name (UPN)format. It can also be specified without a domain simply as user if the domain is instead specified using the domain attribute. On Windows only, if this property is specified, the password property must be specified."
```
Comment (stuartpreston) [inline lib/chef/resource/execute.rb]: `domain\\user` ?
(UPN) format (missing space).

File: lib/chef/resource/execute.rb [hunk 2]
```diff
@@ -55,28 +55,53 @@ def command(arg = nil)
...
+
+      property :domain, String,
+               introduced: "12.21",
+               description: "Windows only: The domain of the user user specified by the user property. Default value: nil. If not specified, the user name and password specified by the user and password properties will be used to resolve that user against the domain in which the system running Chef client is joined, or if that system is not joined to a domain it will resolve the user as a local account on that system. An alternative way to specify the domain is to leave this property unspecified and specify the domain as part of the user property."
```
Comment (stuartpreston) [inline lib/chef/resource/execute.rb]: Is this a fully-qualified domain, or NetBIOS domain name?

File: lib/chef/resource/execute.rb [hunk 3]
```diff
@@ -55,28 +55,53 @@ def command(arg = nil)
...
+
+      property :password, String, sensitive: true,
+               introduced: "12.21",
+               description: "Windows only: The password of the user specified by the user property. Default value: nil. This property is mandatory if user is specified on Windows and may only be specified if user is specified. The sensitive property for this resource will automatically be set to true if password is specified."
 
       # lazy used to set default value of sensitive to true if password is set
       property :sensitive, [ TrueClass, FalseClass ], default: lazy { |r| r.password ? true : false }
 
-      property :elevated, [ TrueClass, FalseClass ], default: false
+      property :elevated, [ TrueClass, FalseClass ], default: false,
```
Comment (stuartpreston) [inline lib/chef/resource/execute.rb]: Does this require a description?

File: lib/chef/resource/file.rb [hunk 4]
```diff
@@ -51,14 +51,29 @@ class File < Chef::Resource
       default_action :create
       allowed_actions :create, :delete, :touch, :create_if_missing
 
-      property :path, String, name_property: true, identity: true
-      property :atomic_update, [ TrueClass, FalseClass ], desired_state: false, default: lazy { |r| r.docker? && r.special_docker_files?(r.path) ? false : Chef::Config[:file_atomic_update] }
-      property :backup, [ Integer, false ], desired_state: false, default: 5
-      property :checksum, [ /^[a-zA-Z0-9]{64}$/, nil ]
-      property :content, [ String, nil ], desired_state: false
+      property :path, String, name_property: true, identity: true,
+               description: "The full path to the file, including the file name and its extension. For example: /files/file.txt. Default value: the name of the resource block See “Syntax” section above for more information. Microsoft Windows: A path that begins with a forward slash (/) will point to the root of the current working directory of the chef-client process. This path can vary from system to system. Therefore, using a path that begins with a forward slash (/) is not recommended."
```
Comment (stuartpreston) [inline lib/chef/resource/file.rb]: Remove smart quotes around "Syntax" ?

File: lib/chef/resource/file.rb [hunk 5]
```diff
@@ -51,14 +51,29 @@ class File < Chef::Resource
...
-      property :manage_symlink_source, [ TrueClass, FalseClass ], desired_state: false
+
+      property :force_unlink, [ TrueClass, FalseClass ], desired_state: false, default: false,
+               description: "How the chef-client handles certain situations when the target file turns out not to be a file. For example, when a target file is actually a symlink. Set to true for the chef-client delete the non-file target and replace it with the specified file. Set to false for the chef-client to raise an error."
+
+      property :manage_symlink_source, [ TrueClass, FalseClass ], desired_state: false,
+               description: "Change the behavior of the file resource if it is pointed at a symlink. When this value is set to true, the Chef client will manage the symlink’s permissions or will replace the symlink with a normal file if the resource has content. When this value is set to false, Chef will follow the symlink and will manage the permissions and content of symlink’s target file. The default behavior is true but emits a warning that the default value will be changed to false in a future version; setting this explicitly to true or false suppresses this warning."
```
Comment (stuartpreston) [inline lib/chef/resource/file.rb]: Is that a smart quote in `symlink's` ?

File: lib/chef/resource/execute.rb (hunk 3 above)
Comment (tas50) [reply lib/chef/resource/execute.rb]: This was missing from the docs site. I created an issue for some of the missing docs

**Pull request merged**